ephios/extra/mixins.py

```diff
@@ -119,3 +121,18 @@ def get_canonical_slug(self):
         object class. In that case, this method will never be called.
         """
         return self.get_object().slug
+
+
+class PermissionFormMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for field in self.fields.values():
+            if isinstance(field, PermissionField) and self.instance.pk is not None:
+                field.set_initial_value(self.permission_target)
+
+    def save(self, commit=True):
+        target = super().save(commit)
+        for key, field in self.fields.items():
+            if isinstance(field, PermissionField) and key in self.changed_data:
+                field.update_permissions(target, self.cleaned_data[key])
+        return target
```

Comment:
As predicted we are getting in trouble here because the permission sets aren't mutually exclusive. Reproduce:

Add a non-staff user to a group. Make that group a management group. Save the form.

In a second form cycle, remove the management permission but leave the hr_management checkbox checked.

The user will then not be able to edit users, even though their group had the HR checkbox checked. When reloading the form, the checkbox will be unchecked.

Here's a possible fix:

1. Don't restrict to `self.changed_data`. Unchanged fields could still provide permissions important because of step 2.
2. First, update permissions for all unchecked Permission fields. Then add all permissions from checked permission fields.

Reply:
If we go that route, we should also split `update_permissions` into `assign_perms` and `remove_perms` to indicate that these are two different and exclusive phases in the PermissionField Livecycle.